be/src/vec/functions/function_regexp.cpp

```diff
@@ -52,12 +51,129 @@
+    }
+
+    DataTypes get_variadic_argument_types_impl() const override {
+        return ParamTypes::get_variadic_argument_types();
+    }
+
+    Status open(FunctionContext* context, FunctionContext::FunctionStateScope scope) override {
+        if (scope == FunctionContext::THREAD_LOCAL) {
+            if (context->is_col_constant(1)) {
+                DCHECK(!context->get_function_state(scope));
+                const auto pattern_col = context->get_constant_col(1)->column_ptr;
+                const auto& pattern = pattern_col->get_data_at(0);
+                if (pattern.size == 0) {
+                    return Status::OK();
+                }
+
+                std::string error_str;
+                std::unique_ptr<re2::RE2> scoped_re;
+                StringRef options_value;
+                if (context->get_num_args() == 4) {
+                    DCHECK(context->is_col_constant(3));
+                    const auto options_col = context->get_constant_col(3)->column_ptr;
+                    options_value = options_col->get_data_at(0);
+                }
+
+                bool st = StringFunctions::compile_regex(pattern, &error_str, StringRef(),
+                                                         options_value, scoped_re);
+                if (!st) {
+                    context->set_error(error_str.c_str());
+                    return Status::InvalidArgument(error_str);
+                }
+                std::shared_ptr<re2::RE2> re(scoped_re.release());
+                context->set_function_state(scope, re);
+            }
+        }
+        return Status::OK();
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        uint32_t result, size_t input_rows_count) const override {
+        size_t argument_size = arguments.size();
+
+        auto result_null_map = ColumnUInt8::create(input_rows_count, 0);
+        auto result_data_column = ColumnString::create();
+        auto& result_data = result_data_column->get_chars();
+        auto& result_offset = result_data_column->get_offsets();
+        result_offset.resize(input_rows_count);
+
+        bool col_const[3];
+        ColumnPtr argument_columns[3];
+        for (int i = 0; i < argument_size; ++i) {
+            col_const[i] = is_column_const(*block.get_by_position(arguments[i]).column);
+        }
+        argument_columns[0] = col_const[0] ? static_cast<const ColumnConst&>(
+                                                     *block.get_by_position(arguments[0]).column)
+                                                     .convert_to_full_column()
+                                           : block.get_by_position(arguments[0]).column;
+
+        default_preprocess_parameter_columns(argument_columns, col_const, {1, 2}, block, arguments);
+
+        // the options have check in FE, so is always const, and get idx of 0
+        StringRef options_value;
+        if (argument_size == 4) {
```

Comment:
the call seems only in 157 line,  `if (col_const[1] && col_const[2])` is true. no need call get_data_at